keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -252,24 +262,26 @@ status:
 
 #### BucketClass
 
-A cluster-scoped custom resource. The bucket class defines a release policy, and specifies driver specific parameters, such as region, bucket lifecycle policies, etc., as well as the name of the driver.  The driver name is used to filter BuckeContenst meant to be handled by a given driver.  In static bucket workflows, the driver name may be empty if the object bucket is defined.
+A cluster-scoped custom resource. The BucketClass defines a release policy, and specifies driver specific parameters, such as region, bucket lifecycle policies, etc., as well as the name of the driver.  The driver name is used to filter BuckeContenst meant to be handled by a given driver.  In static bucket workflows, the driver name may be empty if the object bucket is defined.
 
 ```yaml
 apiVersion: cosi.io/v1alpha1
 kind: BucketClass
 metadata:
   name: 
 provisioner: [1]
-config: string:string [2]
-releasePolicy: {"Delete", "Retain"} [3]
-targetBucket: [5]
-secretName: [6]
+supportedProtocols: [2]
+accessMode: {"ro", "rw"} [3]
+releasePolicy: {"Delete", "Retain"} [4]
+bucketContentName: [5]
+parameters: string:string [6]
 ```
 
 1. `provisioner`: Used by sidecars to filter BucketContent objects 
-1. `config`: object store specific key-value pairs passed to the driver.
+1. `supportedProtocols`: A strings array of protocols the associated object store supports (e.g. swift, s3, gcs, etc.)
+1. `accessModes`: Declares the level of access given to credentials provisioned through this class.
 1.  `releasePolicy`: Prescribes outcome of a Deletion and Revoke events.
     - `Delete`:  the bucket and its contents are destroyed
     - `Retain`:  the bucket and its contents are preserved, only the user’s access privileges are terminated
-- `targetBucket`: A bucket connection endpoint of a brownfield or static bucket.  If provided, assumed to be a GrantAccess() operation.
-- `secretName`: Specifies a  Secret in `cosi-system` namespace containing access credentials required by the driver to operate
\ No newline at end of file
+- `bucketContentName`: (Optional). An admin defined BucketContent representing a Brownfield or Static bucket.  A non-nil value in this field prevents the BucketClass from being used for Greenfield.
+- `parameters`: object store specific key-value pairs passed to the driver.
```

Comment:
Is this change in consideration that not all object stores will need/use a secret, and for those that do the secret name will be a value in `parameters`? I sort of like the idea of "codifying" `secretName` so that we can do api validation.

Reply:
We need to examine what Saad had mentioned about the 3 methods of referencing the secret.